projects/js-packages/publicize-components/src/components/media-section/index.js

```diff
@@ -40,34 +44,55 @@ const ADD_MEDIA_LABEL = __( 'Set Social Image', 'jetpack' );
 const REPLACE_MEDIA_LABEL = __( 'Replace Social Image', 'jetpack' );
 const REMOVE_MEDIA_LABEL = __( 'Remove Social Image', 'jetpack' );
 
-const ALLOWED_MEDIA_TYPES = [ 'image/jpeg', 'image/png' ];
-
 /**
  * Wrapper that handles media-related functionality.
  *
  * @returns {object} The media section.
  */
 export default function MediaSection() {
+	const [ validationError, setValidationError ] = useState( null );
 	const { attachedMedia, updateAttachedMedia } = useAttachedMedia();
+	const { connections } = useSocialMediaConnections();
+	const enabledConnections = connections.filter( connection => connection.enabled );
+
+	const { maxImageSize, allowedImageTypes, getValidationError } = useMediaRestrictions(
+		enabledConnections
+	);
 
 	const mediaObject = useSelect( select =>
 		select( 'core' ).getMedia( attachedMedia[ 0 ] || null, { context: 'view' } )
 	);
-
 	const { mediaWidth, mediaHeight, mediaSourceUrl } = getMediaDetails( mediaObject );
 
+	useEffect( () => {
+		// Removes selected media if connection change results in invalid image
+		if ( ! mediaObject ) {
+			return;
+		}
+		const error = getValidationError( mediaObject.media_details?.filesize, mediaObject.mime_type );
+		if ( error ) {
+			setValidationError( error );
+			updateAttachedMedia( [] );
+		}
+	}, [ updateAttachedMedia, mediaObject, getValidationError ] );
+
 	const onRemoveMedia = useCallback( () => updateAttachedMedia( [] ), [ updateAttachedMedia ] );
 	const onUpdateMedia = useCallback(
 		media => {
 			// allowedTypes doesn't properly disallow uploaded media types.
 			// See: https://github.com/WordPress/gutenberg/issues/25130
-			if ( ! ALLOWED_MEDIA_TYPES.includes( media.mime ) ) {
+			// Do not select media if criteria is not met
+			const error = getValidationError( media.filesizeInBytes, media.mime );
+			if ( error ) {
+				updateAttachedMedia( [] );
+				setValidationError( error );
 				return;
 			}
 
 			updateAttachedMedia( [ media.id ] );
+			validationError && setValidationError( null );
```

Comment:
Do we need to check `validationError` here? It means we have to list it in the dependencies and will change after the call to `setValidationError`, which would cause this to update again.

Reply:
I just wanted to check it to not update it if its already null, but in the end maybe that introduces even more updates! Changed it back.